pkg/kubectl/cmd/run.go

```diff
@@ -146,10 +147,27 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 
 	generatorName := cmdutil.GetFlagString(cmd, "generator")
 	if len(generatorName) == 0 {
+		client, err := f.Client()
+		if err != nil {
+			return err
+		}
+		groupList, err := client.Discovery().ServerGroups()
+		if err != nil {
+			// this cover the cases where old servers do not expose discovery
+			groupList = nil
```

Comment:
That's my idea, I don't have any 1.1 cluster at hand :( Although I can compile 1.1 version and see what I'm gonna get.